signac/contrib/job.py

```diff
@@ -144,7 +146,7 @@ def reset_statepoint(self, new_statepoint):
             else:
                 raise
         # Update this instance
-        self._statepoint = SyncedAttrDict(dst._statepoint._as_dict(), parent=_sp_save_hook(self))
+        self._statepoint._data = dst._statepoint._data
```

Comment:
To make sure I'm understanding, is doing this enforcing that the new statepoint is tied to the exact same file as before by reusing the underlying dict stored in _data?

Reply:
The job's state point is tied to the manifest file through the hook class within the `Job` class constructor. This line is merely copying the in-memory structure. No need to rebuild that.

Reply:
Ah right, the important thing is maintaining the same parent, which is preserved in `_data`. Thanks.